pkg/generate/generate.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package generate scaffolds components to guide a user for component creation
+package generate
+
+import "io/ioutil"
+import "os"
+import "path"
+import "strings"
+
+const componentBuilder = `apiVersion: bundle.gke.io/v1alpha1
+kind: ComponentBuilder
+componentName: etcd-component
+version: 1.0.0
+objectFiles:
+- url: /sample-deployment.yaml
+- url: /sample-service.yaml
+`
+const sampleDeployment = `apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: $DEPLOYMENT_NAME
+  name: helloworld
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: $DEPLOYMENT_NAME
+      name: $DEPLOYMENT_NAME
+    spec:
+      containers:
+        - name: $DEPLOYMENT_NAME
+          image: gcr.io/google-samples/hello-app:2.0
+`
+const sampleService = `apiVersion: v1
+kind: Service
+metadata:
+  name: $SERVICE_NAME
+  labels: 
+    app: $SERVICE_NAME
+spec:
+  type: NodePort
+  ports:
+  - name: $SERVICE_NAME
+    port: 8080
+    targetPort: 8080
+  selector:
+    app: $SERVICE_NAME
+`
+
+// Create scaffolds basic set of files to the filesystem
+func Create(filepath string, name string) error {
+	var writeErr error
+  if err := os.Mkdir(filepath, 0777); err != nil { 
+    writeErr = err 
+    return writeErr
+  }
+
+	if err := ioutil.WriteFile(path.Join(filepath, "etcd-component-builder.yaml"), []byte(componentBuilder), 0666); err != nil {
+    writeErr = err
+  }
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-deployment.yaml"), []byte(strings.Replace(sampleDeployment, "$DEPLOYMENT_NAME", name, -1)), 0666); err != nil {
```

Comment:
How about we use real golang text templates? It's clearer and more robust than ad-hoc variable substitution, imo

(here and throughout).

Reply:
Probably makes sense.  I haven't used golang templates outside of ansible but it should be straight forward. 

I recognize that right not my string substitution can result in pretty broken configs since no validation on input. 

Hopefully golang templates will help constrain that, but either way stylistically makes sense.